### PR TITLE
Add JTS PostGIS data type support

### PR DESCRIPTION
**Pull request by hgschmie**
Support for Postgres PostGIS extension. Based on code by @bchapuis

### CODE REVIEW

**Comment by hgschmie** [comment]
@bchapuis, please review

**Comment by bchapuis** [comment]
@hgschmie Thanks a lot for your help and for the changes, the dedicated module looks good! As mentionned in #2072, I'd love to appear in the commit history for this contribution. I can either cherry pick your last commit in my fork, or you can add me as a co-author in your commit. Either way is fine for me, let me know what you prefer.

```
Co-Authored-By: Bertil Chapuis <bchapuis@gmail.com>
```

**Comment by hgschmie** [comment]
I added you as co-author (and you show up in the commit history). We really appreciate the contribution.

File: postgis/src/test/java/org/jdbi/v3/postgis/PostgisPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+final class PostgisPluginTest {
+
+    static {
+        if (System.getProperty("PG_FULL_IMAGE") == null) {
+            System.setProperty("PG_FULL_IMAGE", "postgis/postgis:13-3.2-alpine");
```
Comment (stevenschlansker) [inline postgis/src/test/java/org/jdbi/v3/postgis/PostgisPluginTest.java:36]: Is it possible that setting this property here affects other tests that might run after? That could introduce test instability depending on order of execution.

File: postgis/src/main/java/org/jdbi/v3/postgis/PostgisCodec.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import static org.locationtech.jts.io.WKBConstants.wkbNDR;
+
+@Alpha
+final class PostgisCodec implements Codec<Geometry> {
+
+    @Override
+    public ColumnMapper<Geometry> getColumnMapper() {
+        return (resultSet, index, context) -> {
+            byte[] bytes = hexStringToByteArray(resultSet.getString(index));
```
Comment (stevenschlansker) [inline postgis/src/main/java/org/jdbi/v3/postgis/PostgisCodec.java:36]: Is it not possible to get the `byte[]` directly via `getBytes`?

**Review by stevenschlansker — COMMENTED** [review]
This is great, thank you for the contribution!

**Pull request merged**